### PR TITLE
Simplify enum generators

### PR DESCRIPTION
**Pull request by knuton**
Explanation copied from [here](https://github.com/prezi/spaghetti/pull/191#issuecomment-107945407):

> I refactored enum value generation to happen in an AST node method. This simplifies generators (language plugins) quite a bit.

> I felt a bit uneasy about doing this on the AST and was considering to introduce an intermediate representation with resolved semantics, but after looking at what other transformations are happening it seemed like overkill. Most translations are quite straightforward and there is little semantic pre-processing. The only thing I really found was method resolution for proxies, which [also happens on the AST](https://github.com/prezi/spaghetti/blob/19eb4a395afdda4163a7d8a583b4d0f28183aebd/spaghetti-core/src/main/java/com/prezi/spaghetti/ast/internal/DefaultInterfaceNode.java#L72-L100).

> Maybe an intermediate representation can be considered in the long run, but I think this is still okay.

@lptr If we merge this into master first, integrating enum generation with the API refactor (#191) will be much easier.

### CODE REVIEW

File: spaghetti-core/src/main/java/com/prezi/spaghetti/ast/EnumNode.java [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 public interface EnumNode extends AnnotatedNode, QualifiedTypeNode, ReferableTypeNode {
 	NamedNodeSet<EnumValueNode> getValues();
+	NamedNodeSet<EnumValueNode> getNormalizedValues();
```
Comment (lptr) [inline spaghetti-core/src/main/java/com/prezi/spaghetti/ast/EnumNode.java]: How about we never return the non-normalized stuff? AST consumers should never be able to access the non-normalized values.

File: spaghetti-core/src/main/java/com/prezi/spaghetti/ast/internal/DefaultEnumNode.java [hunk 2]
```diff
@@ -48,4 +64,65 @@ public void setDocumentation(DocumentationNode documentation) {
...
+					nodes.addInternal(makeValueNodeFrom(enumValueNode, value));
+				}
+				break;
+			case Mixed:
+				throw new AstParserException(
+						this.getLocation().getSource(),
+						"Mixed implicit and explicit entries in enum " + this.getName());
+		}
+		return nodes;
+	}
```
Comment (lptr) [inline spaghetti-core/src/main/java/com/prezi/spaghetti/ast/internal/DefaultEnumNode.java]: Let's do this in the parser, and get rid of the dynamic calculation and the normalized/non-normalized thing.

**Comment by knuton** [comment]
@lptr Please merge if you're happy.

File: spaghetti-core/src/main/java/com/prezi/spaghetti/ast/internal/DefaultEnumNode.java [hunk 3]
```diff
@@ -48,4 +51,5 @@ public void setDocumentation(DocumentationNode documentation) {
 	public NamedNodeSetInternal<EnumValueNode> getValues() {
 		return values;
 	}
+
```
Comment (lptr) [inline spaghetti-core/src/main/java/com/prezi/spaghetti/ast/internal/DefaultEnumNode.java]: This is a non-change with an unnecessary import, please revert.

Comment (lptr) [inline spaghetti-core/src/main/java/com/prezi/spaghetti/ast/internal/DefaultEnumNode.java]: FTFY efdf6860cd8e0283460f837ca7e5d54fda6ec364

**Pull request merged**